src/main/java/Calculator.java

```diff
@@ -1,72 +1,171 @@
 import java.math.BigDecimal;
 
+/**
+ * The Calculator class provides methods to perform arithmetic operations on MoneyValue objects.
+ * It supports addition, subtraction, multiplication, and division of MoneyValue instances.
+ * The class ensures that the operations are performed in the same currency, using a ConverterClient
+ * if necessary to handle currency conversions.
+ */
 public class Calculator {
     private MoneyValueClient mv_;
     private ConverterClient cv_;
 
+    /**
```

Comment:
The @param comments are enough imo since the first two sentences can easily be concluded by the signature.